### DISCUSSION TITLE
How does firejail compare to flatpak with bottles or with flatseal? (firejail vs Bottles vs flatseal)

### DISCUSSION THREAD

**Discussion by mYnDstrEAm** [netblue30/firejail#5418] [Q&A]
How does firejail compare to these in terms of security and other things one may want to compare?

Users recommended these when I asked about [how run Wine securely](https://www.reddit.com/r/linux_gaming/comments/y1zbqb/how_to_run_wine_without_new_security_risks/).

One user also recommended [Bottles](https://usebottles.com/) when I asked about [an option to sandbox Wine with firejail in Lutris](https://github.com/lutris/lutris/issues/4556). At this issue you can find the option to run Wine securely that I currently intend to use and which includes firejail.

What's you take on which would be the best option for securely running Wine? I'd say Bottles or [flatseal](https://github.com/tchx84/Flatseal) (and I don't know if these can be combined or whether one could firejail bottles for combining flatpak and firejail sandboxing) are insufficient for security and not as good as firejail.

Maybe info about this should be added to the documentation, in which case parts of this post & discussion could be moved into an issue.

----

Concerning flathub (and flatpak?), there also is [this security issue](https://github.com/flathub/flathub/issues/1498) relating to integrity (also see the links there).

One user also posted [this criticism of firejail](https://madaidans-insecurities.github.io/linux.html#firejail), has this been addressed before?

**Comment by rusty-snake** [comment]
https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#how-does-it-compare-with-docker-lxc-nspawn-bubblewrap

---

First some clarifications as you compare partly and completely differen things:

- firejail: sandbox
- flatpak: sandbox and app distribution
- bottles: GUI wrapper for wine
- flatseal: GUI configuration for flatpaks sandbox

---

> How does firejail compare to these in terms of security and other things one may want to compare?

What are other things? The only have parts that can be compare.

Here I use the term "security" as protection the host from the app.

**not (yet) complete**:

| / | firejail | flatpak |
| - | -------- | ------- |
| xdg-desktop-portals | No<br>firejailed programs are considered unsandboxed and given full permissions | Yes. |
| private sysroot<sup>1</1> | Optional | Always |
| seccomp | extensive default backlist<br>customizable<br>allowlist supported | minimal default blacklist<br>not customizable<br>can not be disabled |
| capabilities and sandboxing of privileged programs | Yes | No |
| sandboxing of cli programs | Yes | No |
| optional network namespaces sandboxing | Yes | No |
| supported programs | programs installed via apt, dnf, pacman, ...; programs installed from source, archive, ...; AppImages | only flatpaks |
| highly customizable | Yes | No |
| mdwe, xdg-dbus-proxy --call, x11, ... | Yes | No |
| default sandbox configuration | community | flatpak maintainer / developer |


<sup>1</sup>: 

Question: What do you mean with "security", What is you thread model?

---

>Users recommended these when I asked about [how run Wine securely](https://www.reddit.com/r/linux_gaming/comments/y1zbqb/how_to_run_wine_without_new_security_risks/).

Wine in firejail is much more trouble/pain than flatpak / bottles.

> What's you take on which would be the best option for securely running Wine?

Since firejail's sandbox is much more customizable than flatpak's, you have much more control of what is allowed and what not.

However installing wine in you default sysroot bloats it up with i368 libraries if you run on amd64 and don't have any other mulitarch program already installed.

Question: What do you mean with "securely", What is you thread model?

> Concerning flathub (and flatpak?), there also is https://github.com/flathub/flathub/issues/1498 relating to integrity (also see the links there).

Question: What is you thread model?

> One user also posted [this criticism of firejail](https://madaidans-insecurities.github.io/linux.html#firejail), has this been addressed before?

Question: What is you thread model?

To link the section above: https://madaidans-insecurities.github.io/linux.html#flatpak

If you read madaidans-insecurities you fell like you should not use computers. Nothing he writes is wrong on a fact based (if at all outdated, but the project is on GitHub and accepts PRs so this gets fixed rather soon as I see). However he only talks about the black side of the medal.

#4601 and the links at the top.

**Reply by mYnDstrEAm** [reply]
I'll get to the other things later, but I know that these are different things: bottles includes sandboxing via flatpak according to its website, its sandboxing is what is compared with firejail here. I'm not suggesting it's good, I'm just asking for an explanation of the differences and I only forwarded the madaidans link even if I don't find it constructive.
